### PR TITLE
feat: Implements infinite-scroll on Search page

### PR DESCRIPTION
**Pull request by leanhduy**
server: update the Nexus schema.js for  endpoint (add new query type and update the endpoint)

server: add helper method for filtering users based on search term

client: update the variable name on home page for consistency

client: implement the infinite scroll on the search page

client: update the user-card component to remove the refetch code and switch to reload the page using  method of react-router-dom, because refetching is currently conflicted with the way infinite scroll is update and rendering the search page.

Closes P1-58

### CODE REVIEW

File: client/src/pages/search.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import React, { useContext, useEffect, useState } from 'react'
-import { useQuery } from '@apollo/client'
+import React, { useContext, useEffect, useRef, useState } from 'react'
+import { useQuery, useReactiveVar } from '@apollo/client'
```
Comment (pybotlad) [inline client/src/pages/search.js:2]: Where is this hook used?

File: client/src/pages/search.js [hunk 2]
```diff
@@ -10,92 +10,116 @@ import { GET_USERS, GET_USER_BY_ID, searchString } from '../utils'
...
+        }
 
+        return () => {
+            if (fetchMoreRef.current) {
+                observer.unobserve(fetchMoreRef.current)
+            }
+        }
+    }, [users])
+
+    useEffect(() => {
```
Comment (pybotlad) [inline client/src/pages/search.js:68]: What is the purpose of this code? Add comment to explain

File: server/src/schema.js [hunk 3]
```diff
@@ -79,6 +80,22 @@ const FeedQueryResponse = objectType({
   },
 })
 
+// * Custom type for the result of the `feedForYou` query
```
Comment (pybotlad) [inline server/src/schema.js:83]: wrong comment details. Should it be `users` instead of `feedForYou`. Please update.

**Review by pybotlad — CHANGES_REQUESTED** [review]
There are some minor updates need to be made (mostly on the comment parts). Please update them.

File: client/src/container/user-card.js [hunk 4]
```diff
@@ -25,7 +26,7 @@ const UserCard = ({ user, loggedInUser, refetchUsers }) => {
         },
         onCompleted: async (data) => {
             // ? Refetch the list of users
```
[the remark this answers is not in the document]
Comment (leanhduy) [reply client/src/container/user-card.js:28]: Comment updated

File: client/src/container/user-card.js [hunk 5]
```diff
@@ -42,7 +43,7 @@ const UserCard = ({ user, loggedInUser, refetchUsers }) => {
...
```
[the remark this answers is not in the document]
Comment (leanhduy) [reply client/src/container/user-card.js:45]: Comment updated

File: client/src/pages/home.js [hunk 6]
```diff
@@ -24,7 +24,7 @@ const Home = () => {
     // * Client-side GraphQL queries
     // TODO: Replace this with actual authentication context once login is implemented
```
[the remark this answers is not in the document]
Comment (leanhduy) [reply client/src/pages/home.js:25]: Redundant comment removed

File: client/src/pages/search.js (hunk 1 above)
Comment (leanhduy) [reply client/src/pages/search.js:2]: This hook was accidentally added during implementation. Not used anywhere. Removed.

File: client/src/pages/search.js (hunk 2 above)
Comment (leanhduy) [reply client/src/pages/search.js:68]: Explanatory comment added.

File: server/src/schema.js (hunk 3 above)
Comment (leanhduy) [reply server/src/schema.js:83]: Wrong indeed. Updated the comment.

**Review by pybotlad — APPROVED** [review]
I'm happy with the changes
Please merge into `main` branch

**Pull request merged**